### ISSUE TITLE
Add A11y improvements for spotlight experience

### ISSUE THREAD

**Issue by JishnuGoyal** [oppia/oppia-android#4459]
**Is your feature request related to a problem? Please describe.**
Spotlight is completely a visual tool that tells the user “where on the screen a particular element is that should be used now”. And for people using talkback, who can’t see what a spotlight is, are only annoyed/slowed by the spotlight flow, which is totally opposite to what the goal is. 

**Describe the solution you'd like**
Instead of showing spotlights to the talkback users, add appropriate content descriptions on the elements being highlighted by the spotlight. Add a forced announcement for the hint button (when it becomes visible). Suspend the spotlight experience for the users using talkback. Back up this entire functionality using necessary tests.
